### ISSUE TITLE
Code is not generated 

### ISSUE THREAD

**Issue by ahmadmssm** [skydoves/PreferenceRoom#10]

**Comment by skydoves** [comment]
@ahmadmssm 
Hello!
Could you explain more specifically about it?
Is the code not generated from the clone of this repository?

**Comment by ahmadmssm** [comment]
Thanks for you swift response.
I created this class after adding the library dependency `@PreferenceEntity
public class SharedPreference {

    // Share preference keys
    @interface Keys {
        String USER = "user";
    }

    @KeyName(name = "user")
    @TypeConverter(converter = SharedPreferenceTypeConverter.class)
    protected User user;

}`
and didi a clean and build however the code did not genertaed

**Comment by skydoves** [comment]
@ahmadmssm 
You should remove below codes.
```java
@interface Keys {
    String USER = "user";
}
```
Thank you!

**Comment by ahmadmssm** [comment]
Still not solved although i removed the code you mentioned

**Comment by skydoves** [comment]
@ahmadmssm 
Hello, this is the instruction of how to create a base structure.
I think you already create a class `User`.

Firstable, you should create an `UserEntity.class` entity using `@PreferenceEntity` annotation.
```java
@PreferenceEntity(name = "UserEntity")
public class UserEntity {
    @KeyName(name = "user")
    @TypeConverter(converter = SharedPreferenceTypeConverter.class)
    protected User user;
}
```

Second,  build project. `Build` -> `Rebuild Project` on Android Studio IDE.

After build, this library will generates automatically `Preference_UserEntity.class`.

You can reference this [demo](https://github.com/skydoves/PreferenceRoom/tree/master/demo) project.
Thank you for your issue!

**Comment by marconealberto** [comment]
I'm having the same problem. This used to work, until I had to reimport the project in Studio, and now I can't seem to have the class generated. I migrated from 1.1.3 to 1.1.7, just to be sure. This is my class:

```
@PreferenceEntity( "MySettings")
public class MySettings {

	@KeyName("lightMode")
	protected final boolean lightMode = false;

	@KeyName("microphoneService")
	protected final boolean microphoneService = false;

	@KeyName("votingService")
	protected final boolean votingService = false;

	@PreferenceFunction("lightMode")
	public boolean putLightModeFunction(boolean lightMode) {
		return lightMode;
	}

	@PreferenceFunction("microphoneService")
	public boolean putMicServiceFunction(boolean microphoneService) {
		return microphoneService;
	}

	@PreferenceFunction("votingService")
	public boolean putVotingServiceFunction(boolean votingService) {
		return votingService;
	}


}
```

class is not generated. Everytime I run a Rebuild, it abruptly stops saying that it can't find the generate class 

`error: cannot find symbol class Preference_MySettings
`

**Comment by skydoves** [comment]
@marconealberto 
Could you build this [demo](https://github.com/skydoves/PreferenceRoom/tree/master/demo) project?
If the demo project would generate classes, I think the `annotationProcessor` has some problem.

```gradle
annotationProcessor "com.github.skydoves:preferenceroom-processor:1.1.7"
```

**Comment by marconealberto** [comment]
just tried the demo project and it's not working either

`error: cannot find symbol class PreferenceComponent_AppComponent`

but I guess it's something with the import of the project.
I'm sure the demo project works, it's just that something broke on my local machine in the last few weeks and I don't understand what. 
Do you have any pointers? is there a way to mock the creation of the files to check if there's something else that doesn't let me create the classes?

**Comment by skydoves** [comment]
@albert0m 
I think it has code generating error over the 4.0.0 Gradle version.
Because of the 'javaPoet' library has the same [issue](https://github.com/square/javapoet/issues/726).
I released downgraded version `1.1.8`.
Thank you for your issue!
